include/envoy/http/filter.h

```diff
@@ -261,6 +278,16 @@ class StreamFilterBase {
 class StreamDecoderFilter : public StreamFilterBase {
 public:
   /**
+   * This routine is called on filter creation, setting the buffer limit for the
+   * filter.  Filters should abide by these limits unless custom configuration
+   * overrides the limit.  A buffer limit of 0 bytes indicates no limits are applied.
+   *
+   * @param settings supplies the default buffer limit and filter type for this filter.
+   * @return the buffer limit the filter will apply.
+   */
+  virtual uint32_t setDecoderBufferLimit(uint32_t limit) PURE;
```

Comment:
I'm still not crazy about having every filter need to implement this function. Thoughts on ways of making this optional with a sane default? My thought here is to make this a callback function that needs to be called when set*Callbacks() is called on the filter? Open to other thoughts and your general opinion here.